starlette_zipkin/middleware.py

```diff
@@ -29,16 +29,15 @@ def __init__(
     async def dispatch(
         self, request: Request, call_next: RequestResponseEndpoint
     ) -> Response:
-        tracer = get_tracer()
-        if tracer is None:
-            tracer = await init_tracer(self.config)
+        if self.tracer is None:
+            self.tracer = await init_tracer(self.config)
```

Comment:
Here the `_tracer_ctx_var` is not persisted between requests, so we still have to somehow to set it for every subsequent requests

Reply:
oh, ok! Got it. thanks

Reply:
@mchlvl 
Now I really understand how it is supposed to works.

We keep the reference of the tracer in the middleware, and push it in the context on every requests.
And it is supposed to be works exactly like the root span.

As a better solution, one context var could be used for both variable because they are used together.

https://github.com/mchlvl/starlette-zipkin/pull/28/commits/b5ef55c39c5cddb9dfa835ff98b5c9c4cffafa02